Src/Library/Extensions/ExecutorMiddleware.cs

```diff
@@ -12,10 +12,12 @@ internal class ExecutorMiddleware
 {
     private const string authInvoked = "__AuthorizationMiddlewareWithEndpointInvoked";
     private const string corsInvoked = "__CorsMiddlewareWithEndpointInvoked";
+    private readonly IEndpointFactory _epFactory;
     private readonly RequestDelegate _next;
 
-    public ExecutorMiddleware(RequestDelegate next)
+    public ExecutorMiddleware(IEndpointFactory epFactory, RequestDelegate next)
```

Comment:
Do note that since the `ExecutorMiddleware` is effectively a singleton, the developer creating a custom `IEndpointFactory` should also ensure that their implementation runs correctly as a singleton.